File: examples/governance_static_go/identity_static_go/domain/record_entry.go
package domain

import (
	"time"
)

type RecordEntry struct {
	EntityType string `json:"entity_type"`
	EntityId string `json:"entity_id"`
	Action string `json:"action"`
	ActorId string `json:"actor_id"`
	Details string `json:"details"`
}

func (c RecordEntry) CommandName() string { return "RecordEntry" }

func (c RecordEntry) Execute(repo AuditLogRepository) (*AuditLog, *RecordedEntry, error) {
	agg := NewAuditLog(c.EntityType, c.EntityId, c.Action, c.ActorId, c.Details, time.Time{})
	if err := agg.Validate(); err != nil {
		return nil, nil, err
	}
	if err := repo.Save(agg); err != nil {
		return nil, nil, err
	}
	event := RecordedEntry{
		AggregateID: agg.ID,
		EntityType: c.EntityType,
		EntityId: c.EntityId,
		Action: c.Action,
		ActorId: c.ActorId,
		Details: c.Details,
		OccurredAt: time.Now(),
	}
	return agg, &event, nil
}

File: examples/governance/identity_domain/lib/identity_domain/audit_log/commands/record_entry.rb
module IdentityDomain
  class AuditLog
    module Commands
      class RecordEntry
        include Hecks::Command
        emits "RecordedEntry"

        attr_reader :entity_type
        attr_reader :entity_id
        attr_reader :action
        attr_reader :actor_id
        attr_reader :details

        def initialize(
          entity_type: nil,
          entity_id: nil,
          action: nil,
          actor_id: nil,
          details: nil
        )
          @entity_type = entity_type
          @entity_id = entity_id
          @action = action
          @actor_id = actor_id
          @details = details
        end

        def call
          AuditLog.new(
            entity_type: entity_type,
            entity_id: entity_id,
            action: action,
            actor_id: actor_id,
            details: details,
            timestamp: Time.now.to_s
          )
        end
      end
    end
  end
end
